src/common/snippets/include/snippets/op/buffer.hpp

```diff
@@ -13,6 +13,8 @@ namespace op {
 /**
  * @interface Buffer
  * @brief This is a base class for memory storage.
+ *        If Buffer has a parent, the operation is for intermediate data storage - Intermediate type.
```

Comment:
Minor: Let's be precise: `Intermediate` --> `IntermediateMemory`, `Empty` --> `NewMemory`